server/datastore/mysql/apple_mdm.go

```diff
@@ -1164,3 +1164,81 @@ func (ds *Datastore) InsertMDMIdPAccount(ctx context.Context, account *fleet.MDM
 	_, err := ds.writer.ExecContext(ctx, stmt, account.UUID, account.Username, account.Salt, account.Entropy, account.Iterations)
 	return ctxerr.Wrap(ctx, err, "creating new MDM IdP account")
 }
+
+func (ds *Datastore) GetMDMAppleFileVaultSummary(ctx context.Context, teamID *uint) (*fleet.MDMAppleFileVaultSummary, error) {
+	sqlFmt := `
+	SELECT
+	COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND hdek.decryptable = 1
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'applied'
+				AND hmap.operation_type = 'install') THEN
+			1
+		END) AS applied, COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND(hdek.decryptable = 0
+					OR hdek.decryptable IS NULL)
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'applied'
+				AND hmap.operation_type = 'install') THEN
+			1
+		END) AS action_required, COUNT(
+		CASE WHEN EXISTS (
+			SELECT
+				1 FROM host_mdm_apple_profiles hmap
+			WHERE
+				h.uuid = hmap.host_uuid
+				AND hmap.profile_identifier = 'com.fleetdm.fleet.mdm.filevault'
+				AND hmap.status = 'pending'
```

Comment:
A NULL status also indicates that the profile is pending to be installed, that should be `AND (hmap.status IS NULL OR hmap.status = 'pending')`.